mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.h

```diff
@@ -148,6 +148,9 @@ class CNEOBaseCombatWeapon : public CBaseHL2MPCombatWeapon
 	virtual void ItemPostFrame(void);
 
 	virtual void PrimaryAttack(void);
+#ifdef GAME_DLL
+	virtual void DispatchMuzzleParticleEffect(void);
```

Comment:
Probably doesn't need to be virtual if we're rolling our own function. Alternatively could explore overriding `C_BaseAnimating::DispatchMuzzleEffect` and/or `C_BaseAnimating::ProcessMuzzleFlashEvent`.

Reply:
Ok im overriding processmuzzleflashevent now, this allows us to also hide the ent_light when using suppressed weapons, and also now a muzzle flash is created by the client when the muzzleflashevent is triggered.